avm/res/azure-stack-hci/virtual-hard-disk/tests/e2e/defaults/main.test.bicep

```diff
@@ -0,0 +1,196 @@
+      domainFqdn: 'hci.local'
+      domainOUPath: nestedDependencies.outputs.domainOUPath
+      startingIPAddress: '172.20.0.2'
+      endingIPAddress: '172.20.0.7'
+      enableStorageAutoIp: true
+      keyVaultName: nestedDependencies.outputs.keyVaultName
+      networkIntents: [
+        {
+          adapter: ['mgmt']
+          name: 'management'
+          overrideAdapterProperty: true
+          adapterPropertyOverrides: {
+            jumboPacket: '9014'
+            networkDirect: 'Disabled'
+            networkDirectTechnology: 'iWARP'
+          }
+          overrideQosPolicy: false
+          qosPolicyOverrides: {
+            bandwidthPercentage_SMB: '50'
+            priorityValue8021Action_Cluster: '7'
+            priorityValue8021Action_SMB: '3'
+          }
+          overrideVirtualSwitchConfiguration: false
+          virtualSwitchConfigurationOverrides: {
+            enableIov: 'true'
+            loadBalancingAlgorithm: 'Dynamic'
+          }
+          trafficType: ['Management']
+        }
+        {
+          adapter: ['comp0', 'comp1']
+          name: 'compute'
+          overrideAdapterProperty: true
+          adapterPropertyOverrides: {
+            jumboPacket: '9014'
+            networkDirect: 'Disabled'
+            networkDirectTechnology: 'iWARP'
+          }
+          overrideQosPolicy: false
+          qosPolicyOverrides: {
+            bandwidthPercentage_SMB: '50'
+            priorityValue8021Action_Cluster: '7'
+            priorityValue8021Action_SMB: '3'
+          }
+          overrideVirtualSwitchConfiguration: false
+          virtualSwitchConfigurationOverrides: {
+            enableIov: 'true'
+            loadBalancingAlgorithm: 'Dynamic'
+          }
+          trafficType: ['Compute']
+        }
+        {
+          adapter: ['smb0', 'smb1']
+          name: 'storage'
+          overrideAdapterProperty: true
+          adapterPropertyOverrides: {
+            jumboPacket: '9014'
+            networkDirect: 'Disabled'
+            networkDirectTechnology: 'iWARP'
+          }
+          overrideQosPolicy: true
+          qosPolicyOverrides: {
+            bandwidthPercentage_SMB: '50'
+            priorityValue8021Action_Cluster: '7'
+            priorityValue8021Action_SMB: '3'
+          }
+          overrideVirtualSwitchConfiguration: false
+          virtualSwitchConfigurationOverrides: {
+            enableIov: 'true'
+            loadBalancingAlgorithm: 'Dynamic'
+          }
+          trafficType: ['Storage']
+        }
+      ]
+      storageConnectivitySwitchless: false
+      storageNetworks: [
+        {
+          adapterName: 'smb0'
+          vlan: '711'
+        }
+        {
+          adapterName: 'smb1'
+          vlan: '712'
+        }
+      ]
+      subnetMask: '255.255.255.0'
+    }
+  }
+}
+
+resource customLocation 'Microsoft.ExtendedLocation/customLocations@2021-08-31-preview' existing = {
+  scope: resourceGroup
+  name: '${namePrefix}${serviceShort}-location'
```

Comment:
Since this value is duplicated I'd suggest a variable to be defined at the top and use the variable in both the cluster and this existing resource